Rubberduck.VBEEditor/Application/IHostApplication.cs

```diff
@@ -5,10 +5,11 @@ namespace Rubberduck.VBEditor.Application
     public interface IHostApplication : IDisposable
     {
         /// <summary>
-        /// Runs VBA procedure specified by name.
+        /// Runs VBA procedure specified by name. WARNING: The parameter is declared as dynamic to prevent circular referencing.
+        /// This should ONLY be passed a Declaration object.
         /// </summary>
-        /// <param name="qualifiedMemberName">The method to be executed.</param>
-        void Run(QualifiedMemberName qualifiedMemberName);
+        /// <param name="declaration">The Declaration object for the method to be executed.</param>
+        void Run(dynamic declaration);
```

Comment:
The overhead for this should be minimal compared with actually running the unit tests.  Unit testing is the only place ProjectDisplayName is used, and then only in some hosts.  Changing this to dynamic allows only resolving this (which is horrendously expensive and a UI disaster) once per project and once per parse at a maximum, never if unit tests aren't run in an effected host.